cleanlab/classification.py

```diff
@@ -404,105 +556,21 @@ class label.
-                )
-
-            self.sample_weight = np.ones(np.shape(labels_cleaned))
-            for k in range(self.K):
-                sample_weight_k = 1.0 / max(self.noise_matrix[k][k], 1e-3)  # clip sample weights
-                self.sample_weight[labels_cleaned == k] = sample_weight_k
-
-            if self.verbose:
-                print("Fitting final model on the clean data ...")
-            self.clf.fit(
-                x_cleaned, labels_cleaned, sample_weight=self.sample_weight, **self.clf_final_kwargs
-            )
-        else:
-            if self.verbose:
-                print("Fitting final model on the clean data ...")
-            # This is less accurate, but best we can do if no sample_weight.
-            self.clf.fit(x_cleaned, labels_cleaned, **self.clf_final_kwargs)
-
-        return self.clf
-
-    def predict(self, *args, **kwargs):
-        """Returns a vector of predictions.
-
-        Parameters
-        ----------
-        X : :obj:`np.array` of shape (n, m)
-          The test data as a feature matrix."""
-
-        return self.clf.predict(*args, **kwargs)
-
-    def predict_proba(self, *args, **kwargs):
-        """Returns a vector of probabilities P(true_label=k)
-        for each example in X.
+            print(f"Identified {np.sum(self.label_issues_mask)} datapoints with label issues.")
 
-        Parameters
-        ----------
-        X : :obj:`np.array` of shape (n, m)
-          The test data as a feature matrix."""
-
-        return self.clf.predict_proba(*args, **kwargs)
-
-    def score(self, X, y, sample_weight=None):
-        """Returns the clf's score on a test set X with labels y.
-        Uses the model/clf's default scoring function.
-
-        Parameters
-        ----------
-        X : :obj:`np.array` of shape (n, m)
-          The test data as a feature matrix.
-
-        y : :obj:`np.array<int>` of shape (n,) or (n, 1)
-          The test classification labels as an array.
-
-        sample_weight : :obj:`np.array<float>` of shape (n,) or (n, 1)
-          Weights each example when computing the score / accuracy."""
-
-        if hasattr(self.clf, "score"):
-
-            # Check if sample_weight in clf.score(). Not compatible with Python 2.
-            if "sample_weight" in inspect.getfullargspec(self.clf.score).args:
-                return self.clf.score(X, y, sample_weight=sample_weight)
-            else:
-                return self.clf.score(X, y)
-        else:
-            return accuracy_score(
-                y,
-                self.clf.predict(X),
-                sample_weight=sample_weight,
-            )
+        return self.label_issues_mask
 
     def get_label_issues(self):
-        """Accessor. Returns `self.label_issues_mask` computed via `filter.find_label_issues()`"""
+        """Accessor. Returns `self.label_issues_mask` if previously already computed."""
 
         return self.label_issues_mask
 
```

Comment:
`_process_label_issues_kwargs` has several changes needed:
* it should not set global constants. I'd move MIN_EXAMPLES_PER_CLASS = 10 as a class global (why is it 10 btw? please clarify in a comment) and then just pass this global wherever its needed in a function call and remove this from here
* the docstring should be "Throws an error if you pass in a kwargs argument to filter.find_label_issues that is not supported by the LearningWithNoisyLabels.find_label_issues() function"
* the comment below the docstring should say "# Different defaults to use for LearningWithNoisyLabels.find_label_issues() vs filter.find_label_issues():"
* The confident_joint is being set here to `self.confident_joint` but if you follow the code in `LearningWithNoisyLabels.find_label_issues()` you'll see it gets set again afterwards. Instead the logic in `LearningWithNoisyLabels.find_label_issues` needs to be changed so that if the confident joint exists, you just compute `noise_matrix` and `inverse_noise_matrix` using `count.estimate_latent(labels, confident_joint)` (this would go just before the part where you compute the `pred_probs` if they don't exist.
* given all these issues i strongly recommend removing this function entirely and just set `return_indices_ranked_by=""` and `multi_label=False` directly when `filter.find_label_issues()` is called (we are going to add multi_label support anyway so that part will get removed soon too.)

Reply:
fixed the second, third, and fourth bullets and marked the last bullet as a TODO.